### PR TITLE
feat: Added Preloader

### LINKED ISSUE #106: Add Preloader
**Describe the new feature**
A preloader is an essential feature for enhancing user experience by indicating the progress of loading content. It helps in keeping the users engaged while the main content is being fetched or rendered. Implementing a preloader can significantly improve the perceived performance of our application.

**Describe the possible approach**
**Design and Appearance:**
1. A simple and visually appealing animation.
2. Should match the overall design language of the application.
3. Customizable options for color, size, and style to fit different themes.

**Functionality:**
1. Should display during the loading of heavy components or data fetching processes.
2. Must cover the main content area that is being loaded.
3. Option to show the preloader globally (entire screen) or locally (specific component).

**Performance:**
1. Lightweight and optimized to prevent additional load times.
2. Should not interfere with the loading process of the main content.

**Implementation:**
1. Should be implemented as a reusable component.
2. Easy to integrate with different parts of the application.
3. Must include appropriate documentation for usage and customization.

**Describe any alternatives**
1. Skeleton Screens: Instead of a preloader animation, skeleton screens can be used to display a placeholder for content that is loading. This approach can be less distracting and provide a more seamless experience.
2. Progress Bars: Another alternative is to use a progress bar that shows the percentage of content loaded. This can provide more detailed feedback to users about the loading process.
3. Lazy Loading Indicators: For sections of content that load as the user scrolls, implementing lazy loading indicators can also serve as an alternative to a full preloader.

**Additional context**
My Previous Work:-
https://github.com/dvjsharma/Drawn2Shoe/assets/146121869/175ff458-54a0-4b1c-b9a0-ce6d4b71aa08

### PR DESCRIPTION
**Pull request by AsmitaMishra24**
## Description
Added Preloader.

Issue No.:
Fixes #106 

Screenshot:-
https://github.com/dvjsharma/Drawn2Shoe/assets/146121869/1853f739-b28b-43d6-8e81-64bc745b50e4

### CODE REVIEW

File: client/src/components/Preloader/Preloader.json
Comment (dvjsharma) [inline client/src/components/Preloader/Preloader.json:1]: Is there any CDN which provides this `json` over links?

Comment (dvjsharma) [inline client/src/components/Preloader/Preloader.json:1]: Is there any CDN which provides this `json` over links?

File: client/src/components/Preloader/Preloader.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        } else {
+            document.body.classList.remove('preloader-active');
+        }
+    }, [loading]);
+
+    if (!loading) return null;
+
+    return (
+        <div className={`preloader ${fadeIn ? "fade-in" : "fade-out"}`}>
+            <Lottie className="preloader-animation" animationData={preloaderAnimation} />
```
Comment (dvjsharma) [inline client/src/components/Preloader/Preloader.jsx:32]: Please provide screenshots since this is a UI PR.

**Review by dvjsharma — CHANGES_REQUESTED** [review]
Some questions, PTAL.

**Comment by AsmitaMishra24** [comment]
@dvjsharma, I have added a preloader.json from the lottiefiles website, but I can also design one using CSS. The design will be simple.

**Comment by AsmitaMishra24** [comment]
@dvjsharma, I will design it with CSS...in 2-3 days

**Comment by dvjsharma** [comment]
The current design looks good to me. Just see if there are alternate on link delivery for the json file, else we can go with the current format.

**Comment by AsmitaMishra24** [comment]
The alternative methods for delivering the JSON file are quite similar to the current format, so the current format is good to go. Alternatively, we can create a link for the JSON file.

**Comment by dvjsharma** [comment]
Yep, that’s what I’m concerned about. See if you can replace the JSON with a deliverable link, so we don’t have to manage the JSON file in code itself.

**Comment by AsmitaMishra24** [comment]
> Yep, that’s what I’m concerned about. See if you can replace the JSON with a deliverable link, so we don’t have to manage the JSON file in code itself.

Ok, I will let you know in 2 days.

**Comment by AsmitaMishra24** [comment]
@dvjsharma, I've searched this, and it appears that adding JSON files via links is only feasible in traditional html and js only. In React, this method isn't applicable. Most resources suggest directly adding the JSON file to the project.

For reference:-
When working with traditional HTML and JavaScript, it's possible to dynamically load JSON files using <script> tags and the src attribute. However, in React applications, we leverage modern JavaScript modules and utilize the import statement to include JSON files directly into our JavaScript code. This is due to React's modular structure and how assets are managed during the build process.

**Comment by AsmitaMishra24** [comment]
Could you upgrade the level. I received level 2 for the same work in other repos.

**Comment by dvjsharma** [comment]
There are methods to inject scripts, but the current implementation appears satisfactory. I think level 1 is good enough for this feature request. We'll see if this requires an upgrade.
Thanks for your contribution.

EDIT: Upgrading to level 2

**Review by dvjsharma — APPROVED** [review]
Changes looks good.

**Pull request merged**